### DISCUSSION TITLE
USB boot - idea discarded

### DISCUSSION THREAD

**Discussion by stiebel** [ltsp/ltsp#323] [Q&A]
Hi, 
My name is Roland and I have a special issue:
I used to use ltsp for the last 10 years at my school.
Now we have to switch to windows and even the pxe-boot is reserved for this.
I am allowed to use Linux and I have a server for that, but I am not allowed to install
anything on the clients. The boot-options on the PCs are open, so I can use this.
So using an USB live system is one option, but I cannot adminstrate 100 USB Sticks.

So my idea is the following:
All that stuff that is usually done by PXE/DHCP/TFTP schould move to an USB stick.
This should be only a (large) couple of megabytes.
Then the main system is loaded via NFS from the server, giving me the chance to only maintain one system.
User mangement and filestorage is also done by the server.
Only at major system updates I have to upgrade the usb-sticks, I will think about that later, but this will be fast,
due to its small size.

But how do I do this?
Can I transfer ltsp to USB-boot?
Or should I go a different way?

I would appreciate your help !
(probably I am not the only one, who's linux urge is badgered by similar restrictions)

Greetings,
Roland

**Comment by maccadia** [comment]
(I guess this issue will be moved to [discussion](https://github.com/ltsp/ltsp/discussions))

You can install ipxe on a USB stick see for example https://gist.github.com/AdrianKoshka/5b6f8b6803092d8b108cda2f8034539a

You will need to add an [ipxe script](https://ipxe.org/scripting), something like :

```
#!ipxe
set net0/next-server [YOUR LTSP SERVER]
chain tftp://${next-server}/ltsp/ltsp.ipxe
```

I also successfully added an iPXE EFI boot option to Windows hosts. iPXE is stored in the EFI partition alongside Windows bootloader.

I can provide some help to accomplish this, unfortunately I don't have step by step instructions ready.

**Comment by stiebel** [comment]
Hi MAccadia,
thanks for your post.
Yes I know, this is an option.
I discuss this at a different point with alkis allready.
Here I really want to use only NFS on the server and all the other stuff on USB.

**Comment by alkisg** [comment]
Hi Roland,

MAccadia is 100% correct.

When there's a bug in LTSP code, we use issues: https://github.com/ltsp/ltsp/issues
When we want to discuss something, we use discussions: https://github.com/ltsp/ltsp/discussions

An LTSP server provides DHCP, TFTP, NFS and SSHFS services.
You want to avoid DHCP, i.e. you want to use a [typical installation](https://ltsp.org/docs/installation/), but then when you configure dnsmasq, you want to disable proxydhcp:

```shell
ltsp dnsmasq -p0
```

You do **not** want "only NFS". What about SSHFS, where will the user /home come from?
TFTP provides the kernel and initrd, which in Ubuntu 20.04 are around 100 MB. Why would you want to put these in the USB stick, so that you have to update them all the time?

The way that MAccadia proposed is the best, you just put an ipxe with an embedded script in the USB sticks, i.e. just 1 MB, which does normal DHCP, and then has a hardcoded IP for the LTSP server.

**Comment by stiebel** [comment]
OK, you convinced me.
Thank you both!

I will follow the ipxe way.

Roland
